### PR TITLE
Fix #712

### LINKED ISSUE #712: `volta list` does not display defaults when project version is set
The design for `volta list` specified that we would output both the default version from the user's toolchain and the version set in a local package:

```sh
$ volta list all --format=plain
runtime node@v12.2.0
runtime node@v11.9.0
runtime node@v10.15.3 (default)
runtime node@v8.16.0 (current @ ~/node-only/package.json)
```

We currently only print the `(current @ ...)` when in a package with one set, however:

```sh
$ volta list all --format=plain
runtime node@v12.2.0
runtime node@v11.9.0
runtime node@v10.15.3
runtime node@v8.16.0 (current @ ~/node-only/package.json)
```

As expected, this affects both "plain" and "human" formats.

### PR DESCRIPTION
**Pull request by 2hu12**
Fix #712: `volta list` does not display defaults when project version is set

Example output of release version
image

Example output of local dev version
image

### CODE REVIEW

File: src/command/list/toolchain.rs [hunk 1]
```diff
@@ -55,6 +55,17 @@ impl Lookup {
         default_platform: Option<&PlatformSpec>,
         version: &Version,
     ) -> Source {
+        let default_or_none = default_platform
+            .clone()
```
Comment (charlespierce) [inline src/command/list/toolchain.rs:59]: suggestion: It looks like this clone is redundant. It was there before, so if we're refactoring it makes sense to also clean that up.

```suggestion
```

**Comment by 2hu12** [comment]
> Thanks for this @2hu12! Out of curiosity, do you have an example of the updated output when there is a project version?
> 
> I don't think this actually fixes #712, as that issue is around Packages, which are loaded differently than the runtime or package mangers. The code for loading the list of packages (and their sources) is actually in `command/list/mod.rs`. That said, this looks like a nice fix that improves the output for runtimes / package managers in the case that the version is inherited.

File: src/command/list/toolchain.rs (hunk 1 above)
Comment (2hu12) [reply src/command/list/toolchain.rs]: For sure, thanks!

**Review by charlespierce — APPROVED** [review]
Hi @2hu12 I completely dropped the ball on this, my apologies! You were right to be confused by my comment, I think I was looking at #711 when I made that reference to packages.

As a bonus, I believe this also fixes #761 and #922 as they all have the same root cause.

**Pull request merged**